Paintroid/src/main/java/org/catrobat/paintroid/MainActivity.java

```diff
@@ -296,14 +298,66 @@ protected void onDestroy() {
 		PaintroidApplication.savedPictureUri = null;
 		PaintroidApplication.saveCopy = false;
 
-		PaintroidApplication.commandManager.storeCommandLists();
 		PaintroidApplication.commandManager.setInitialized(false);
+		PaintroidApplication.commandManager.resetAndClear(false);
+
+		PaintroidApplication.colorPickerInitialColor = Color.BLACK;
 
 		IndeterminateProgressDialog.getInstance().dismiss();
 		ColorPickerDialog.getInstance().dismiss();
 		super.onDestroy();
 	}
 
+	@Override
+	public void onConfigurationChanged(Configuration newConfig) {
+		super.onConfigurationChanged(newConfig);
+
+		if (newConfig.orientation == 2)
+			getWindow().addFlags(WindowManager.LayoutParams.SCREEN_ORIENTATION_CHANGED);
+		else
+			getWindow().clearFlags(WindowManager.LayoutParams.SCREEN_ORIENTATION_CHANGED);
+
+		ColorPickerDialog.init(this);
+		IndeterminateProgressDialog.init(this);
+		BrushPickerView.init(this);
+
+		setContentView(R.layout.main);
+		initActionBar();
+		mInputMethodManager =  (InputMethodManager)getSystemService(Context.INPUT_METHOD_SERVICE);
+
+		PaintroidApplication.orientation = getResources().getConfiguration().orientation;
+		PaintroidApplication.drawingSurface = (DrawingSurface) findViewById(R.id.drawingSurfaceView);
+		PaintroidApplication.perspective = new Perspective(
+				((SurfaceView) PaintroidApplication.drawingSurface).getHolder());
+		mDrawingSurfaceListener = new DrawingSurfaceListener();
+		mBottomBar = new BottomBar(this);
+		mTopBar = new TopBar(this, PaintroidApplication.openedFromCatroid);
+		mLayerSideNav = (NavigationView) findViewById(R.id.nav_view_layer);
+		mLayerSideNavList = (ListView) findViewById(R.id.nav_layer_list);
+		mLayersAdapter = new LayersAdapter(this, PaintroidApplication.openedFromCatroid,
+				PaintroidApplication.drawingSurface.getBitmapCopy());
+
+		int colorPickerBackgroundColor = PaintroidApplication.colorPickerInitialColor;
+		ColorPickerDialog.getInstance().setInitialColor(colorPickerBackgroundColor);
+
+		PaintroidApplication.drawingSurface
+				.setOnTouchListener(mDrawingSurfaceListener);
+
+		PaintroidApplication.drawingSurface.resetBitmap(LayerListener.getInstance().getCurrentLayer().getImage());
+		PaintroidApplication.perspective.resetScaleAndTranslation();
+		PaintroidApplication.currentTool
+				.resetInternalState(Tool.StateChange.NEW_IMAGE_LOADED);
+		PaintroidApplication.isPlainImage = true;
+		PaintroidApplication.isSaved = false;
+		PaintroidApplication.savedPictureUri = null;
```

Comment:
why setting the three variables above? PaintroidApplication does not change, so I think these three lines can be removed